server/src/test/java/org/elasticsearch/index/shard/ShardGetServiceTests.java

```diff
@@ -242,6 +242,8 @@ public void testGetFromTranslog() throws IOException {
         assertNull(getResult);
         var lastUnsafeGeneration = engine.getLastUnsafeSegmentGenerationForGets();
         assertThat(lastUnsafeGeneration, greaterThan(0L));
+        // last unsafe generation is set to last committed gen after the refresh triggered by realtime get
+        assertThat(engine.getLastCommittedSegmentInfos().getGeneration(), equalTo(lastUnsafeGeneration));
```

Comment:
nit: maybe 
`assertThat(lastUnsafeGeneration, equalTo(engine.getLastCommittedSegmentInfos().getGeneration()));` 
instead of having two asserts on L244 and 246.